scene/animation/animation_tree.cpp

```diff
@@ -838,11 +838,11 @@ void AnimationTree::_setup_animation_player() {
 	// Using AnimationPlayer here is for compatibility. Changing to AnimationMixer needs extra work like error handling.
 	AnimationPlayer *player = Object::cast_to<AnimationPlayer>(get_node_or_null(animation_player));
 	if (player) {
-		if (!player->is_connected(SNAME("caches_cleared"), callable_mp(this, &AnimationTree::_setup_animation_player))) {
```

Comment:
Unless I missed something, you are connecting the same signals to the same node below. Why?